### PR TITLE
Fix table function analyze bug

### PR DESCRIPTION
**Pull request by HangyuanLiu**

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java [hunk 1]
```diff
@@ -541,6 +542,11 @@ public Void visitFunctionCall(FunctionCallExpr node, Scope scope) {
                         node.getParams().isStar() ? "*" : Joiner.on(", ")
                                 .join(Arrays.stream(argumentTypes).map(Type::toSql).collect(Collectors.toList())));
             }
+
+            if (fn instanceof TableFunction) {
+                throw unsupportedException("Does not support TableFunction not used in expression");
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java:547]:
```suggestion
                throw unsupportedException("Don't support TableFunction not used in expression");
```

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java [hunk 2]
```diff
@@ -597,6 +597,11 @@ private Expr analyzeJoinUsing(List<String> usingColNames, Scope left, Scope righ
     private Scope analyzeFrom(SelectStmt node, AnalyzeState analyzeState, Scope scope) {
         Relation sourceRelation = null;
         TableRef lastTableRef = null;
+
+        if (node.getTableRefs().size() == 1 && node.getTableRefs().get(0) instanceof FunctionTableRef) {
+            throw unsupportedException("Does not support TableFunction not used in conjunction with lateral join");
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java:602]:
```suggestion
            throw unsupportedException("Don't support TableFunction not used in conjunction with lateral join");
```

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java [hunk 3]
```diff
@@ -597,6 +597,11 @@ private Expr analyzeJoinUsing(List<String> usingColNames, Scope left, Scope righ
     private Scope analyzeFrom(SelectStmt node, AnalyzeState analyzeState, Scope scope) {
         Relation sourceRelation = null;
         TableRef lastTableRef = null;
+
+        if (node.getTableRefs().size() == 1 && node.getTableRefs().get(0) instanceof FunctionTableRef) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java:601]: What if node.getTableRefs().size() is 2, and one fo table ref is `FunctionTableRef `?

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java:601]: Analyze success

**Review by kangkaisen — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**